### DISCUSSION TITLE
"Utility Hardpoints": An extra step in sub progression and variation

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#6180] [Ideas and suggestions]
After getting weapons progression, it would be a good idea to use the same system for "utility", with a one-time purchase adding particular item assemblies to fill an empty slot on the ship: 

**The utility hardpoint**
![Capture](https://user-images.githubusercontent.com/73229309/123511005-17000500-d67f-11eb-90e2-fb9851732176.JPG)

**Examples of potential locations for hardpoints in Vanilla subs**
![UtilityHardpointShips](https://user-images.githubusercontent.com/73229309/123511023-2c752f00-d67f-11eb-8351-ce8b21c7b9c2.jpg)

If we're getting sub variants later, especially factions/pirates, using hardpoints for randomized sub layouts would be add some variety to encounters

**Comment by soledad630** [comment]
Honestly great idea, there's 2 slight add about it tho:

Spare suits should have oxygen tanks as well.

Ships' existing interior modules that can already fit into this pattern should be also considered a installed utility hardpoint (like already installed turrets). And some of the interiors should be modified into fitting into the pattern.

**Comment by esbenheick** [comment]
Every way players can claim ownership of specific parts of the submarine is an improvement. I think this is a wonderful idea. May I suggest that Drone Control can be added to such a utility hardpoint as well ? Drones could then be bought and added to any upwards or downwards hatch. While docked, it would be an extra "layer" of airlock, so you climb through the drone / shuttle to get out of the sub, if it is attached. 

It would add a way to get access to shuttles for people who like the vanilla subs but cant man those humongous vanilla subs that sport them.

**Reply by ghost** [reply]
I 100% agree with purchasable shuttles / drones, but they should be a selectable option on docking ports instead of available for all docking ports.

**Comment by Hinny** [comment]
This is a great idea. This could be expanded to include a variety of utility sets almost indefinitely.

More examples:
- Bunk beds with drawers
- Operating bed
- Table and chairs
- Crate shelves (x2)
- Backup battery installation:
  - 2 x Batteries
  - Junction box
  - Power switch
- Armory:
  - Safe
  - Weapon holders (x2)
- Emergency installation:
  - Diving suit locker
  - Emergency cabinet
  - Fire extinguisher
- Pump station:
  - Small Locker
  - Bilge pump
  - Emergency cabinet

Maybe wiring is not included and must be setup in game by an Electrician to create more wiring gameplay (since we already have such mechanic implemented in the game, but currently not that important/used).

**Reply by mLuby** [reply]
Excellent suggestion—this would _really_ add to the game, especially if players can publish "Pods" similar to Outpost Modules.

Here are a bunch of Pods I created to illustrate a range of possibilities. The size is standardized to fit one Unit Load Device.
![Pods](https://user-images.githubusercontent.com/2483420/140788684-34293a2a-5b14-44f4-9076-91c8dab7f135.png)

Next, I retrofitted a vanilla Orca with 2 standardized "Pod Bays" labelled A and B.
![Pod Orca wiki](https://user-images.githubusercontent.com/2483420/140788677-686131a7-e4ae-4924-816d-d5c66bdb13ac.png)

Lastly, in-game I docked pods of my choice into those bays, in this case a Cargo Pod in Pod Bay A and a Medbay Pod in Pod Bay B. Now the crew of the orca can interact with the additional storage of the unit load device and the medical fabricator and hospital bed of the Medbay. (Note: unsure what effect buying "Replace Lost Shuttles" at a station has on this setup.)
![Pod Orca in-game](https://user-images.githubusercontent.com/2483420/140788666-09a16b69-beee-458c-9f80-dd8b1ebc73ea.png)

Related proposal in that it deals with progression and changing submarine installations: https://github.com/Regalis11/Barotrauma/discussions/7212

**Comment by Prolbo** [comment]
Interesting concept, I like it. (even if have much simpler things on my mind: bases for super capacitors and junction boxes / batteries)

**Comment by Ivanzyfer** [comment]
It would be great if this feature was added at least in the future DLC, it would go well with building your own outpost. For example, you need to build a shipyard first to change the submarine.
